Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -342,6 +348,42 @@ void WebsiteDataStore::unregisterProcess(WebProcessProxy& process)
     m_processes.remove(process);
 }
 
+String WebsiteDataStore::migrateMediaKeysStorageIfNecessary(const String& directory)
+{
+    if (directory.isEmpty())
+        return emptyString();
+
+    static constexpr ASCIILiteral versionName = "v1"_s;
+    auto versionDirectory = FileSystem::pathByAppendingComponent(directory, versionName);
+    auto saltPath = FileSystem::pathByAppendingComponent(versionDirectory, "salt"_s);
+    m_mediaKeysStorageSalt = valueOrDefault(FileSystem::readOrMakeSalt(saltPath));
+
+    auto originDirectoryNames = FileSystem::listDirectory(directory);
+    // Migrate existing data to new version directory.
+    for (auto name : originDirectoryNames) {
+        if (name == versionName)
+            continue;
```

Comment:
Here we just move to the next item in the directory. Shouldn't we `break` here if we find the new file indicating it was migrated?

Reply:
Existence of new version directory does not indicate successful migration of all data, e.g. process crashes during the migration and only some origin directories are moved to new version directory.